metpy/io/upperair.py

```diff
@@ -162,17 +162,33 @@ def parse(fobj):
         -------
         dict of information used by :func:`get_upper_air_data`
         """
+        def to_float(s):
+            # Remove all whitespace and replace empty values with NaN
+            if not s.strip():
+                s = 'nan'
+            return float(s)
+
         # Skip the row of dashes and column names
         for _ in range(2):
             fobj.readline()
 
         # Parse the actual data, only grabbing the columns for pressure, T/Td, and wind
-        cols = (0, 2, 3, 6, 7)
         unit_strs = ['degC' if u == 'C' else u
                      for u in fobj.readline().decode('ascii').split()]
 
-        # Skip 2 header lines -- 1 for '----' and 1 for partial line at 1000mb
-        p, t, td, direc, spd = np.genfromtxt(fobj, usecols=cols, skip_header=2, unpack=True)
+        # Skip last header row of dashes
+        fobj.readline()
+
+        # Initiate lists for variables
+        arr_data = []
+
+        # Read all lines of data and append to lists
+        for row in fobj.readlines():
+            arr_data.append((to_float(row[0:7]), to_float(row[14:21]),
```

Comment:
To address the empty starting points, how about we we skip all levels where we have no data:
```python
level = to_float(row[0:7])
values = (to_float(row[14:21]), to_float(row[21:28]), to_float(row[42:49]),  to_float(row[49:56]))
if any(values):
    arr_data.append((level, ) + values)
```